pkg/rpc/client/client.go

```diff
@@ -53,6 +58,11 @@ type Options struct {
 	CACert         string
 	DialTimeout    time.Duration
 	RequestTimeout time.Duration
+	// RequestIDsGenerator is a function for RPC request IDs generation. If
+	// Client is to be used in a multi-thread mode, RequestIDsGenerator must
+	// return unique request ID each time it is called. If not set, then the
+	// default generator will be used.
+	RequestIDsGenerator func() int
```

Comment:
Why don't you just use a synchronized counter? Atomic increment is enough for this, you don't need IDs to be random, you only need them to be unique.